Makefile

```diff
@@ -4,12 +4,14 @@ MAKE := make
 RM := rm
 INSTALL := install
 GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags)
+INC_DIR = include
+# DEPS = $(INC_DIR)/common.h
```

Comment:
If you are going to split it out, you really do need to add the dependencies to the Makefile. If it is just a handful of files, this should be easy to manage.

Reply:
No, I don't want to handle dependencies in the makefile, How to do automatic dependency management?

Reply:
This is an advanced Makefile topic and I think we are better off doing it manually for a small number of files. You can use wildcards, but they are more error prone than you imagine: https://stackoverflow.com/questions/28358957/makefile-wildcard-dependencies

Reply:
Better, but still incomplete answer: https://www.math.colostate.edu/~yzhou/computer/writemakefile.html#Using%20Wildcards

Reply:
grrr.... everyone recommends them: https://pages.tacc.utexas.edu/~eijkhout/istc/html/gnumake.html#Wildcards

I cannot easily find the general issues I've encountered, so I guess let's use wildcards and maybe we won't rediscover them.